src/services/SyncService.js

```diff
@@ -298,25 +305,12 @@ class SyncService {
 		})
 	}
 
-	async sendRemainingSteps(queue) {
-		if (queue.length === 0) {
+	sendRemainingSteps() {
+		if (!this.outbox.hasUpdate()) {
```

Comment:
Should probably be `!this.#outbox.hasUpdate()`.

Reply:
Thanks! Plus the hasUpdate is a getter - so need to also remove the `()`.

Interesting sideeffect of this was that `sendRemainingSteps` erred out and `disconnect` never got called in the editor so the connection stayed up even though the viewer was closed. Made that more robust as well.